### PR TITLE
Add recipe for pymt_ecsimplesnow

### PR DESCRIPTION
**Pull request by mdpiper**
This PR adds a conda recipe for the `pymt_ecsimplesnow` package, which brings the (Fortran) ECSimpleSnow model into Python as a plugin for the [PyMT modeling framework](https://pymt.readthedocs.io).

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mdpiper** [comment]
@conda-forge/staged-recipes This recipe is ready for review when someone has a moment. Thank you!

File: recipes/pymt_ecsimplesnow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - bmi-fortran=1.2
```
Comment (chrisburr) [inline recipes/pymt_ecsimplesnow/meta.yaml]: Is `bmi-fortran` supposed to be in all three types of requirements? Naively I would expect only `host` and `run`.
```suggestion
```

File: recipes/pymt_ecsimplesnow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - python
+    - pip
+    - cython
+    - numpy
+    - model_metadata
+    - bmi-fortran=1.2
```
Comment (chrisburr) [inline recipes/pymt_ecsimplesnow/meta.yaml]: Regardless there need to be a space before the version constraint:
```suggestion
    - bmi-fortran =1.2
```

File: recipes/pymt_ecsimplesnow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - ecsimplesnow
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - bmi-fortran=1.2
```
Comment (chrisburr) [inline recipes/pymt_ecsimplesnow/meta.yaml]:
```suggestion
    - bmi-fortran =1.2
```

**Comment by mdpiper** [comment]
@chrisburr Thanks for the suggestions. I'm still getting used to the `host` requirements.

**Comment by chrisburr** [comment]
Yeah it takes a little getting used to. In general `build` can be thought of as "dependencies which only a developer need to interact with" such as `make`, compilers, `sed`, `cmake`, `autotools`...

**Pull request merged**